packages/platforms/delta/arm64/tn48m-dn/src/arm64_delta_tn48m_dn/module/src/psui.c

```diff
@@ -364,12 +364,12 @@ onlp_psui_info_get(onlp_oid_t id, onlp_psu_info_t* info)
             if (info->mvin == 0)
                 info->status |= ONLP_PSU_STATUS_FAILED;
         }
-
-        // TODO : auto detect AC / DC type
-        // Just do a guess
-        info->caps |= _psu_vin_type_guess(info->mvin);
     }
 
+    // TODO : auto detect AC / DC type
+    // Just do a guess
+    info->caps |= _psu_vin_type_guess(info->mvin);
```

Comment:
Where is `info->mvin` set? The check `if (info->caps & ONLP_PSU_CAPS_VIN) {` seems to fail and indicate, that the capability does not exist, right?

Reply:
> Where is `info->mvin` set?

The  `info->mvin`  is default set to zero here:
https://github.com/dentproject/dentOS/blob/e40146e89e39c37a0c4315da34240dadce4fadd9/packages/platforms/delta/arm64/tn48m/src/arm64_delta_tn48m/module/src/psui.c#L350


>  The check `if (info->caps & ONLP_PSU_CAPS_VIN) {` seems to fail and indicate, that the capability does not exist, right?

Right. If we didn't set the value to info->caps, the type attribute will become Unknown.
```
PSU 2
    Description: PSU2
    State: Present
    Status: Running 
    Model: DPS-200PB-204 A
    SN: GQHD194900004
    Type: Unknown
```

Which should be AC by default:
```
 PSU 1
   Description: PSU1
   State: Present
   Status: Running
   Model: DPS-200PB-204 A
   SN: GQHD194900020
   Type: AC
```

Reply:
Yes, I understand, but the check should succeed, shouldn’t it. Could you please figure out, why the capability is not set in https://github.com/dentproject/dentOS/pull/52/files#diff-7041a58922d52fbb218f70b5254df5bf9dcf6330efc9b3883dbf11fffa9984feR343?

    ///////////////////////////////////////////////////////////////
    // Get caps
    if (psu->vin_path && plat_os_file_is_existed(psu->vin_path))
        info->caps |= ONLP_PSU_CAPS_VIN;

Reply:
Do the configure pathes in `.vin_path` for each PSU exist? For example:

    /sys/bus/i2c/devices/0-005b/psu_v_in

Reply:
Hi Paul,
I think you got the point.
Only the PSU modules (dps-920ab) of tn48m-poe support the PMBUS command to read the information of vin/vout/iin/iout/pin/pout. Other PSU modules of other platforms (tn48m/tn4810m/tn48m2) do not have such capability to get these information. So that is why if we do```_psu_vin_type_guess(info->mvin)``` in ```if (info->caps & ONLP_PSU_CAPS_VIN) {``` will cause the problem.

Reply:
That’s something to note in the commit message. ;-) If you know, PMBUS is not supported, then autodetection is not needed. I suggest to extend the structure `plat_psu_t`.  Add a member `.vin` or `.vin_fixed` or something, and then extend the check further below.

Reply:
> That’s something to note in the commit message. ;-) If you know, PMBUS is not supported, then autodetection is not needed. I suggest to extend the structure `plat_psu_t`. Add a member `.vin` or `.vin_fixed` or something, and then extend the check further below.

I feel it will be simple to set PSU capability to AC by default if the vin path is not exist, which also means the PSU model does not support PMBUS commands.